app/Services/Model/Imp/SummitService.php

```diff
@@ -2028,6 +2028,34 @@ public function removeModeratorFromPresentation(int $current_member_id, int $spe
         });
     }
 
+    /**
+     * @param int $speaker_id
+     * @param int $presentation_id
+     * @param int $new_order
+     * @return Presentation
+     * @throws EntityNotFoundException
+     * @throws ValidationException
+     */
+    public function updateSpeakerOrderInPresentation(int $speaker_id, int $presentation_id, int $new_order): Presentation {
+        return $this->tx_service->transaction(function () use ($speaker_id, $presentation_id, $new_order) {
+
+            $presentation = $this->event_repository->getById($presentation_id);
+            if (is_null($presentation))
+                throw new EntityNotFoundException(sprintf("presentation %s not found", $presentation_id));
+
+            if (!$presentation instanceof Presentation)
+                throw new EntityNotFoundException(sprintf("presentation %s not found", $presentation_id));
+
+            $speaker = $this->speaker_repository->getById(intval($speaker_id));
+            if (is_null($speaker) || !($speaker instanceof PresentationSpeaker))
+                throw new EntityNotFoundException(sprintf('speaker %s not found', $speaker_id));
+
+            $presentation->updateSpeakerOrder($speaker, $new_order);
```

Comment:
we should check that new_order is not equal to current order to avoid unnecessary updates

Reply:
done